### PR TITLE
Rename ContentsKey to ContentKey and Contents to Content

### PR DESCRIPTION
**Pull request by nastra**
This is the first part of #2584 and is a pure rename of `ContentsKey` to ContentKey and `Contents` to `Content` with additional renames of variables and docs in Java/Python/Markdown files.

### CODE REVIEW

File: model/src/main/java/org/projectnessie/model/IcebergTable.java [hunk 1]
```diff
@@ -47,19 +47,19 @@
     title = "Iceberg table global state",
     description =
         "Represents the global state of an Iceberg table in Nessie. An Iceberg table is globally "
-            + "identified via its unique 'Contents.id'.\n"
+            + "identified via its unique 'Content.id'.\n"
```
Comment (dimas-b) [inline model/src/main/java/org/projectnessie/model/IcebergTable.java:50]: nit: just `content ID` perhaps? What's the meaning of the `.` there? It does not look like a property path :thinking:

File: python/tests/test_nessie_cli.py [hunk 2]
```diff
@@ -209,16 +209,16 @@ def test_tag() -> None:
...
     _make_commit("commit.expected.contents", _new_table("test_expected_contents"), "dev", message="commit 1")
-    # the second commit will use the Contents of the first one as "expected contents"
+    # the second commit will use the Content of the first one as "expected contents"
     _make_commit("commit.expected.contents", _new_table("test_expected_contents"), "dev", message="commit 2")
 
 
 @pytest.mark.vcr
 def test_commit_no_expected_state() -> None:
-    """Test making two commit without any expected Contents, i.e. using DeltaLakeTable."""
+    """Test making two commit without any expected Content, i.e. using DeltaLakeTable."""
```
Comment (dimas-b) [inline python/tests/test_nessie_cli.py:221]: nit: `two commits` with an `s`

File: site/docs/develop/spec.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ user facing applications like Web UIs.
 Nessie defines a [REST API](rest.md) (OpenAPI) and implementations for [Java](java.md)
 and [Python](python.md).
 
-## Contents in Nessie
+## Content in Nessie
```
Comment (dimas-b) [inline site/docs/develop/spec.md:16]: nit: `Content Managed by Nessie`?

File: ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap [hunk 4]
```diff
@@ -107,3 +107,17 @@ exports[`Commit log renders 2`] = `
...
+  <div
+    class="EmptyView card-header"
+  >
+    <span
+      class="EmptyView__message"
+    >
+      Nothing to show
+    </span>
+  </div>
+</DocumentFragment>
```
Comment (dimas-b) [inline ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap:122]: Why this change? Is it related to the "content" renaming?

**Review by dimas-b — COMMENTED** [review]
The renaming of `Contents` to `Content` makes sense to me overall. I think the related code and messages are more readable now (at least from my non-native English perspective :) )

Some fairly minor comments added.

File: clients/client/src/main/java/org/projectnessie/client/api/NessieApiV1.java [hunk 5]
```diff
@@ -30,7 +30,7 @@
   /** Get details for the default reference. */
   Branch getDefaultBranch() throws NessieNotFoundException;
 
-  GetContentsBuilder getContents();
+  GetContentBuilder getContent();
```
[the remark this answers is not in the document]
Comment (nastra) [reply clients/client/src/main/java/org/projectnessie/client/api/NessieApiV1.java:33]: I believe `Map<ContentKey, Content> response = api.getContent().key(a).key(b).key(c).refName("foo").get();` is more readable, so I would prefer to name it `getContent()`

File: model/src/main/java/org/projectnessie/model/Content.java [hunk 6]
```diff
@@ -26,10 +26,10 @@
 import org.eclipse.microprofile.openapi.annotations.media.Schema;
 import org.immutables.value.Value;
 
-/** Abstract implementation of contents within Nessie. */
+/** Abstract implementation of content within Nessie. */
```
[the remark this answers is not in the document]
Comment (nastra) [reply model/src/main/java/org/projectnessie/model/Content.java:29]: sgtm, changed

File: python/tests/test_nessie_cli.py (hunk 2 above)
Comment (nastra) [reply python/tests/test_nessie_cli.py:221]: fixed

File: model/src/main/java/org/projectnessie/model/IcebergTable.java (hunk 1 above)
Comment (nastra) [reply model/src/main/java/org/projectnessie/model/IcebergTable.java:50]: not sure, there are also a few other places that refer to it like that. Would probably make sense to update this in a separate PR if we really want it

File: ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap (hunk 4 above)
Comment (nastra) [reply ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap:122]: it was auto-generated after running `./mvnw clean install`

File: clients/client/src/main/java/org/projectnessie/client/api/NessieApiV1.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply clients/client/src/main/java/org/projectnessie/client/api/NessieApiV1.java:33]: SGTM

File: ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap (hunk 4 above)
Comment (dimas-b) [reply ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap:122]: could you move it to a separate PR (before this one)? This PR is already huge and these UI changes are not related...

File: model/src/main/java/org/projectnessie/api/ContentApi.java [hunk 7]
```diff
@@ -20,63 +20,64 @@
...
+   * @param hashOnRef hash on {@code ref} to retrieve the content for, translates to {@code HEAD},
    *     if missing/{@code null}
-   * @param request the {@link ContentsKey}s to retrieve
-   * @return list of {@link MultiGetContentsResponse.ContentsWithKey}s
+   * @param request the {@link ContentKey}s to retrieve
+   * @return list of {@link ContentWithKey}s
    * @throws NessieNotFoundException if {@code ref} or {@code hashOnRef} does not exist
    */
-  MultiGetContentsResponse getMultipleContents(
+  MultiGetContentResponse getMultipleContents(
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply model/src/main/java/org/projectnessie/api/ContentApi.java:76]: or simply `GetContentsResponse`?

[the remark this answers is not in the document]
Comment (nastra) [reply model/src/main/java/org/projectnessie/api/ContentApi.java:76]: done

File: site/docs/develop/spec.md (hunk 3 above)
Comment (nastra) [reply site/docs/develop/spec.md]: updated

File: ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap (hunk 4 above)
Comment (dimas-b) [reply ui/src/Explore/__snapshots__/CommitLog.test.tsx.snap:122]: yet, it's not a big deal :)

**Review by dimas-b — APPROVED** [review]
LGTM :+1:

**Review by omarsmak — APPROVED** [review]
LGTM 👍🏽 .

P/S: I will need to rebase my PR after this gets merged
